src/main/java/com/statful/client/framework/springboot/registry/StatfulMetricRegistry.java

```diff
@@ -0,0 +1,190 @@
+
+                    return new Tag() {
+                        @Override
+                        public String getKey() {
+                            return splitTag[0];
+                        }
+
+                        @Override
+                        public String getValue() {
+                            return splitTag[1];
+                        }
+                    };
+                })
+                .collect(Collectors.toList());
+    }
+
+    protected void publish() {
+        this.getMeters().forEach((meter) -> {
+            if (meter instanceof Timer) {
+                this.writeTimer((Timer) meter);
+            } else if (!(meter instanceof FunctionTimer) && !(meter instanceof DistributionSummary) && !(meter instanceof TimeGauge)) {
+                if (meter instanceof Gauge) {
+                    this.writeGauge((Gauge) meter);
+                } else if (meter instanceof Counter) {
+                    this.writeCounter((Counter) meter);
+                } else if (meter instanceof FunctionCounter) {
+                    this.writeCounter((FunctionCounter) meter);
+                } else if (meter instanceof LongTaskTimer) {
+                }
+            }
+
+        });
+    }
+
+    protected void writeCounter(FunctionCounter counter) {
+        Meter.Id meterId = counter.getId();
+
+        com.statful.client.domain.api.Tags tags = new com.statful.client.domain.api.Tags();
+        meterId.getTags().forEach((tag) -> tags.putTag(tag.getKey(), tag.getValue()));
+
+        this.statfulClient.counter(meterId.getName(), Double.valueOf(counter.count()).intValue()).with().tags(tags).send();
+    }
+
+    protected void writeCounter(Counter counter) {
+        Meter.Id meterId = counter.getId();
+
+        com.statful.client.domain.api.Tags tags = new com.statful.client.domain.api.Tags();
+
+        meterId.getTags().forEach((tag) -> tags.putTag(tag.getKey(), tag.getValue()));
+
+        this.statfulClient.counter(meterId.getName(), (new Double(counter.count())).intValue()).with().tags(tags).send();
+    }
+
+    protected void writeGauge(Gauge gauge) {
+        Meter.Id meterId = gauge.getId();
+        com.statful.client.domain.api.Tags tags = new com.statful.client.domain.api.Tags();
+
+        meterId.getTags().forEach((tag) -> tags.putTag(tag.getKey(), tag.getValue()));
+
+        this.statfulClient.gauge(meterId.getName(), gauge.value()).with().tags(tags).send();
+    }
+
+    protected void writeTimer(Timer timer) {
+        HistogramSnapshot histogramSnapshot = timer.takeSnapshot();
+        Meter.Id meterId = timer.getId();
+        com.statful.client.domain.api.Tags tags = new Tags();
+
+        meterId.getTags().forEach((tag) -> tags.putTag(tag.getKey(), tag.getValue()));
+
+        long value = (new Double(histogramSnapshot.mean(this.getBaseTimeUnit()))).longValue();
+
+        this.statfulClient.timer(meterId.getName(), value).with().tags(tags).send();
```

Comment:
@senzzzi I had some thoughts regarding this one. Since it already applies a mean I thought it might make sense to call aggregatedTimer() instead but since you were already using it like this I decided to leave it.